packages/frontend/src/components/Election/Results/BallotDataExport.tsx

```diff
@@ -0,0 +1,97 @@
+import { CSVLink } from 'react-csv';
+import { useState, useEffect } from 'react';
+import { Election } from '@equal-vote/star-vote-shared/domain_model/Election';
+import { AnonymizedBallot, Ballot } from '@equal-vote/star-vote-shared/domain_model/Ballot';
+import { useGetAnonymizedBallots } from '~/hooks/useAPI';
+import MenuItem from "@mui/material/MenuItem";
+import BorderAll from '@mui/icons-material/BorderAll';
+import DataObject from '@mui/icons-material/DataObject';
+import { MenuButton } from '~/components/MenuButton';
+
+
+
+
+
+
+interface Props {
+    election: Election;
+}
+
+export const BallotDataExport = ({ election }: Props) => {
+    const [csvData, setCsvData] = useState([]);
+    const [csvHeaders, setCsvHeaders] = useState([]);
+	const { data, error, isPending, makeRequest } = useGetAnonymizedBallots(election.election_id);
```

Comment:
I'm realizing that this could be a expensive call for larger elections, can you have it wait to do the call until the download is confirmed?

Reply:
Yeah, I was having a super hard time getting it to work with the hook, so I ended up just doing it outside. But I'll keep working on it